endpoint.bal

```diff
@@ -0,0 +1,372 @@
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/http;
+import ballerina/log;
+
+# Object for Google Contacts configuration.
+#
+# + oauth2Config - OAuth client configuration
+# + secureSocketConfig - HTTP client configuration
+public type GoogleContactsConfiguration record {
+    http:BearerTokenConfig | http:OAuth2RefreshTokenGrantConfig oauth2Config;
+    http:ClientSecureSocket secureSocketConfig?;
+};
+
+# Google Contacts Client.
+#
+# + googleContactClient - The HTTP Client
+@display {label: "Google People API Client", iconPath: "GooglePeopleLogo.png"}
+public client class Client {
+    public http:Client googleContactClient;
+
+    public isolated function init(GoogleContactsConfiguration googleContactConfig) returns error? {
+        http:ClientSecureSocket? socketConfig = googleContactConfig?.secureSocketConfig;
+        self.googleContactClient = check new (BASE_URL, {
+            auth: googleContactConfig.oauth2Config,
+            secureSocket: socketConfig
+        });
+    }
+
+    # Fetch all from "Other Contacts".
+    # 
+    # + readMasks - Restrict which fields on the person are returned
+    # + return - Stream of `PersonResponse` on success else an `error`
+    @display {label: "List OtherContacts"}
+    isolated remote function listOtherContacts(@display {label: "Read Masks"} OtherContactMasks[] readMasks, 
+                                      @display {label: "Optional query parameters"} ContactListOptions? options = ()) 
+                                      returns @tainted @display {label: "Stream of PersonResponses"} stream<PersonResponse>|error {
+        string path = LIST_OTHERCONTACT_PATH;
+        http:Request request = new;
+        string pathWithReadMasks = prepareUrlWithReadMasks(path, readMasks);
+        PersonResponse[] persons = [];
+        return getOtherContactsStream(self.googleContactClient, persons, pathWithReadMasks, options);
+    }
+
+    # Create a contact.
+    # 
+    # + createContact - Record of type of `CreatePerson`
+    # + personFields - Restrict which fields on the person are returned
+    # + return - `PersonResponse` on success else an `error`
+    @display {label: "Create Contact"}
+    isolated remote function createContact(@display {label: "Contact details"} Person createContact, 
+                                  @display {label: "Person Fields"} ContactMasks[]? personFields = ()) returns 
```

Comment:
What supposed to be in `personFields` from user perspective? In Choreo UI how we can input an array of ContactMasks elements?

Reply:
Filter options should be specified. Currently it should be provided as array of strings